File: backend/app/models/curriculum.py
"""Pydantic models for Curriculum entity."""

from pydantic import BaseModel, Field
from typing import List

class CurriculumBase(BaseModel):
    title: str = Field(..., example="Mathematics 10th Grade")
    description: str | None = Field(None, example="Basic algebra and geometry")
    grade_level: int = Field(..., example=10)
    subjects: List[str] = Field(..., example=["Algebra", "Geometry", "Trigonometry"])

class CurriculumCreate(CurriculumBase):
    pass

class CurriculumUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    grade_level: int | None = None
    subjects: List[str] | None = None

class Curriculum(CurriculumBase):
    id: int

    class Config:
        orm_mode = True

File: lib/models/curriculum.dart
class Curriculum {
  final int id;
  final String title;
  final String description;
  final int gradeLevel;
  final List<String> subjects;

  Curriculum({
    required this.id,
    required this.title,
    required this.description,
    required this.gradeLevel,
    required this.subjects,
  });

  factory Curriculum.fromJson(Map<String, dynamic> json) {
    return Curriculum(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      gradeLevel: json['grade_level'],
      subjects: List<String>.from(json['subjects'] ?? []),
    );
  }
}
